src/macho.rs

```diff
@@ -296,44 +324,142 @@ pub struct DyldCacheHeader<E: Endian> {
     /// e.g. "dyld_v0    i386"
     pub magic: [u8; 16],
     /// file offset to first dyld_cache_mapping_info
-    pub mapping_offset: U32<E>, // offset: 0x10
+    pub mapping_offset: U32<E>,
     /// number of dyld_cache_mapping_info entries
-    pub mapping_count: U32<E>, // offset: 0x14
-    /// file offset to first dyld_cache_image_info
-    pub images_offset: U32<E>, // offset: 0x18
-    /// number of dyld_cache_image_info entries
-    pub images_count: U32<E>, // offset: 0x1c
+    pub mapping_count: U32<E>,
+    /// UNUSED: moved to imagesOffset to prevent older dsc_extarctors from crashing
+    pub images_offset_old: U32<E>,
+    /// UNUSED: moved to imagesCount to prevent older dsc_extarctors from crashing
+    pub images_count_old: U32<E>,
```

Comment:
This is a breaking change, and furthermore the old names are now the names of fields in a different position, so this could silently break users. Which of the new fields that you have added do you actually need?

Reply:
This is how apple rolls. I've updated this structure to match the latest dyld source. Whenever interacting with apple structures like this I would not rely on structure locations to stay the same, but write wrappers around where they are used that handles the change in their usage in dyld. I figure that's roughly why the functions in the read module exist?